### PR TITLE
gke-private Terraform module refactor

### PR DESCRIPTION
**Pull request by lnovara**
The aim of this PR is to refactor the `gke-private` Terraform module mainly to adhere to [Terraform naming conventions](https://www.terraform.io/docs/extend/best-practices/naming.html) and to allow flexible handling of node pools.

I am not particularly happy about how I ended up handling `node_taints` in the module but I feel is the only way to do it with Terraform 0.11, comments are more than welcome. The 0.12 might improve this a little using the [`object`](https://www.terraform.io/docs/configuration/types.html#object-) type, but I think that we have to give up defaults in that case.

I have summarised the changes introduced by this PR in the [release notes document](https://github.com/sighupio/fury-kubernetes-gke/blob/feature/gke-private-refactor/docs/releases/v1.0.0.md).

### CODE REVIEW

**Comment by angelbarrera92** [comment]
I have to take a view deeper, but looks good. Just a question. Do you know if the module is fully compatible with terraform 0.12? We should move to 0.12 as 0.11 is going to be deprecated soon.

File: modules/gke-private/README.md [hunk 1]
```diff
@@ -25,50 +25,52 @@ terraform {
     bucket  = "customer-terraform"
     prefix  = "sighup"
   }
-  required_version = ">= 0.11.11"
+  required_version = "~> 0.11.14"
```
Comment (angelbarrera92) [inline modules/gke-private/README.md]: Can we move this module to be compatible with terraform 0.12?

File: modules/gke-private/README.md [hunk 2]
```diff
@@ -25,50 +25,52 @@ terraform {
...
-  node-number = 1
-  node-type = "n1-standard-2"
-  node-os = "COS"
-  infra-node-number = 0
-  subnetwork-node-cidr = "10.1.0.0/16"
-  subnetwork-pod-cidr = "10.3.0.0/16"
-  subnetwork-svc-cidr = "10.5.0.0/16"
-  bastion-ssh-enabled = true
-  bastion-vpn-enabled = true
+  kube-master-version    = "1.11.6-gke.6"
```
Comment (angelbarrera92) [inline modules/gke-private/README.md]: It's a really old version. We should consider, at least, using a variable to specify the kubernetes version. What do you think about it?

File: docs/releases/v1.0.0.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+- Added `daily_maintenance_time` input variable
+- Explicitely set `use_ip_aliases = true` in `ip_allocation_policy` block for
+  future-proofing
+- Explicitely set `provider = "CALICO"` in `network_policy` block otherwise the
+  default was `PROVIDER_UNSPECIFIED`
+- Improved `google_container_node_pool` resource creation handling
+- Added taints handling in `google_container_node_pool` resources
+
+### Migration
+Rename the following input variables in your `module` stanzas.
```
Comment (ralgozino) [inline docs/releases/v1.0.0.md]: Is `stanzas` the right word here? I've never heard it before in this context.

File: docs/releases/v1.0.0.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+| `bastion-ssh-enabled`    | `bastion_ssh_enabled`    |
+| `bastion-vpn-enabled`    | `bastion_vpn_enabled`    |
+| `kube-node-version`      | `kube_node_version`      |
+| `master-authorized-cidr` | `master_authorized_cidr` |
+| `subnetwork-master-cidr` | `subnetwork_master_cidr` |
+| `subnetwork-node-cidr`   | `subnetwork_node_cidr`   |
+| `subnetwork-pod-cidr`    | `subnetwork-pod-cidr`    |
+| `subnetwork-svc-cidr`    | `subnetwork_svc_cidr`    |
+
+Remove the following input variable from your `module` stanzas:
```
Comment (ralgozino) [inline docs/releases/v1.0.0.md]: Same as before

File: docs/releases/v1.0.0.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+```bash
+$ terraform plan -target=module.<YOUR_GKE_PRIVATE_MODULE_NAME>.google_container_cluster.main
+```
+you should see that the Terraform plan wants just to update your existing
+cluster and if you are fine with the plan
+```bash
+$ terraform apply -target=module.<YOUR_GKE_PRIVATE_MODULE_NAME>.google_container_cluster.main
+```
+
+Than, to switch to the new node pools avoiding downtime, execute the following
```
Comment (ralgozino) [inline docs/releases/v1.0.0.md]: "Than" -> "Then"

File: docs/releases/v1.0.0.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+```bash
+$ terraform plan -target=module.<YOUR_GKE_PRIVATE_MODULE_NAME>.google_container_node_pool.pool
+```
+you should see that the Terraform plan wants just to create new node pools and
```
Comment (ralgozino) [inline docs/releases/v1.0.0.md]: "Terraform plan wants just to create"  -> "Terraform plan just wants to create", personally, I'll drop the "just".

File: modules/gke-private/README.md (hunk 2 above)
Comment (ralgozino) [reply modules/gke-private/README.md]: This are just a sample `main.tf` inside the README instructions, you are supposed to take it as reference and set all the parameteres according to what you want to achieve. Maybe we can add a note saying that though.

File: modules/gke-private/README.md (hunk 1 above)
Comment (ralgozino) [reply modules/gke-private/README.md]: In my opinion the migration to terraform 0.12.x is out of the scope of this PR. It's a broader work that definitely should be done, but in another thread and it's part of a bigger task.

**Comment by lnovara** [comment]
Closing this PR since it's superseded by the awesome [sighupio/fury-gke-installer](https://github.com/sighupio/fury-gke-installer).

**Pull request closed without merging**